lab_3_generate_by_ngrams/main.py

```diff
@@ -34,13 +35,30 @@ def _tokenize(self, text: str) -> Optional[tuple[str, ...]]:
 
         Args:
             text (str): Original text
-
         Returns:
             tuple[str, ...]: Tokenized text
 
         In case of corrupt input arguments, None is returned.
         In case any of methods used return None, None is returned.
         """
+        if not isinstance(text, str) or not text or not any(char.isalpha() for char in text):
+            return None
+        self.text = text
+        splited_text = text.split()
+        list_of_tokens = []
+        for word in splited_text:
+            current_word = ''.join(char.lower() for char in word if char.isalpha())
+            if current_word:
+                list_of_tokens += current_word
+                list_of_tokens.append(self._end_of_word_token)
+
+        if text[-1].isalnum():
```

Comment:
why do you do it?

Reply:
I have not fully understood the question, unfortunately. I first made a list from the text with the split method, then made all words lower if they consist of letters and added them to the empty list, appending the end of word token...
If that's what you were asking about....